odoo/modules/module.py

```diff
@@ -426,7 +429,15 @@ def get_modules_with_version():
 def adapt_version(version):
     serie = release.major_version
     if version == serie or not version.startswith(serie + '.'):
+        base_version = version
         version = '%s.%s' % (serie, version)
+    else:
+        base_version = version[len(serie) + 1:]
+
+    if not re.match(r"^[0-9]+\.[0-9]+$", base_version):
+        raise ValueError(f"Invalid version {base_version!r}. Modules should have a version in format `x.y`.")
```

Comment:
Hi @KangOl 

This requirement seems a bit ...impactful. For example, in OCA we use v.w.x.y.z format, where v is number version of Odoo. Thus, this change forces us to change our format number for our modules :S

Reply:
Hello @KangOl is there a fix planned for this or should I submit a PR ?
We develop importable modules for SAAS and it is preventing us to use our usual version number format which is the same as the OCA.

Reply:
It has been handled. This case should now work.
Do you have an example of a failing version?

Reply:
Hello, I just tested today in 16.3 and it is failing if you have a module in the addons-path that has more than 3 numbers in its version.
That being said I am no longer able to reproduce it with an importable module as this week we managed to import them successfully in odoo.com.

Reply:
More than three is not supported. That was part of the aim of this change.